From 17b3ff826d3820e5df224220fdc0a6af5c3286ad Mon Sep 17 00:00:00 2001
From: Christopher Gervais <chris@ergonlogic.com>
Date: Thu, 7 Mar 2013 15:55:31 +0000
Subject: [PATCH] Issue #1931514 by ergonlogic: Ensure git_branches exists prior to running settings hook.

---
 devshop_projects/inc/nodes.inc |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/devshop_projects/inc/nodes.inc b/devshop_projects/inc/nodes.inc
index 3d093ea..3ceda8f 100644
--- a/devshop_projects/inc/nodes.inc
+++ b/devshop_projects/inc/nodes.inc
@@ -36,7 +36,7 @@ function devshop_projects_node_info() {
  */
 function devshop_projects_nodeapi(&$node, $op, $a3 = null) {
 
-  if ($node->type == 'project') {
+  if ($node->type == 'project' && isset($node->git_branches)) {
     // Settings are added here, since $git_branches hadn't been added to the
     // node yet in devshop_projects_load().
     $settings = module_invoke_all('devshop_project_settings', $node);
-- 
1.7.2.5

